From eff2d8e00e137bc00cd221daabd967b14c8a12e5 Mon Sep 17 00:00:00 2001
From: Thor Thayer <thor.thayer@linux.intel.com>
Date: Tue, 19 Dec 2017 10:33:01 -0600
Subject: [PATCH 03/25] arm64: dts: stratix10: fix SPI settings

commit  1f10cc4ddb2718339ee83108dae5edf693047adf from
https://github.com/altera-opensource/linux-socfpga.git

[backport 'commit 51edd22628ea ("FogBugz #514871-2: arm64:
    dts: stratix10: fix SPI settings")']

Correct the SPI Master node settings.

Signed-off-by: Thor Thayer <thor.thayer@linux.intel.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi b/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
index 5dd4bcd..6b8d2f9 100644
--- a/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
+++ b/arch/arm64/boot/dts/altera/socfpga_stratix10.dtsi
@@ -543,7 +543,7 @@
 				spi_m_clk: spi_m_clk {
 					#clock-cells = <0>;
 					compatible = "altr,socfpga-s10-gate-clk";
-					clocks = <&l4_mp_clk>;
+					clocks = <&l4_main_clk>;
 					clk-gate = <0xa4 9>;
 				};
 
@@ -739,7 +739,10 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0xffda4000 0x1000>;
-			interrupts = <0 101 4>;
+			interrupts = <0 99 4>;
+			clocks = <&spi_m_clk>;
+			resets = <&rst SPIM0_RESET>;
+			reg-io-width = <4>;
 			num-chipselect = <4>;
 			bus-num = <0>;
 			status = "disabled";
@@ -750,7 +753,10 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0xffda5000 0x1000>;
-			interrupts = <0 102 4>;
+			interrupts = <0 100 4>;
+			clocks = <&spi_m_clk>;
+			resets = <&rst SPIM1_RESET>;
+			reg-io-width = <4>;
 			num-chipselect = <4>;
 			bus-num = <0>;
 			status = "disabled";
-- 
2.8.1

